src/sync/sync-utils.ts

```
@@ -65,25 +51,53 @@ export const findOrStartSync = async (
-	if (!targetTasks?.length) {
-		return;
+const resetRepoTaskStatusAndCursor = async (subscription: Subscription, syncType: SyncType, targetTasks?: TaskType[]) => {
+
+	const repoSyncTasks = (targetTasks || []).filter(t => t !== "repository");
+	const targetTaskIsEmpty = (targetTasks || []).length === 0;
+	const hasSubscriptionRepositoryTask = (targetTasks || []).includes("repository");
+	const hasRepoSyncTask = repoSyncTasks.length > 0;
+
+	if (syncType == "full") {
+		//For full sync
+		if (targetTaskIsEmpty) {
+			// Didn't provide any target tasks for a full sync, so remove all state as we're starting anew
+			await subscription.update({ totalNumberOfRepos: null, repositoryCursor: null, repositoryStatus: null });
+			await RepoSyncState.deleteFromSubscription(subscription);
+		} else {
+			if (hasRepoSyncTask) {
+				await resetTargetedTasks({ subscription, repoSyncTasks, resetCursor: true });
+			}
+			if (hasSubscriptionRepositoryTask) {
+				await subscription.update({ totalNumberOfRepos: null, repositoryStatus: null, repositoryCursor: null });
+			}
+			await RepoSyncState.update({ failedCode: null }, { where: { subscriptionId: subscription.id } });
+		}
+	} else {
+		//For partial sync
+		if (hasRepoSyncTask) {
+			await resetTargetedTasks({ subscription, repoSyncTasks, resetCursor: false });
+		}
+		if (hasSubscriptionRepositoryTask) {
+			await subscription.update({ repositoryStatus: null });
+		}
 	}
+};
+
+
+type ResetTargetedTasksParam = { subscription: Subscription, resetCursor: boolean, repoSyncTasks: TaskType[] };
+const resetTargetedTasks = async ({ subscription, resetCursor, repoSyncTasks }: ResetTargetedTasksParam): Promise<void> => {
 
-	// Reset RepoSync states - target tasks: ("pull" | "commit" | "branch" | "build" | "deployment")
-	// Full sync resets cursor and status
-	// Partial sync only resets status (continues from existing cursor)
-	const repoSyncTasks = targetTasks.filter(task => task !== "repository");
```

Could we throw an exception if there's "repository" in the array?